haystack/modeling/training/base.py

```diff
@@ -281,7 +281,8 @@ def train(self):
         # With early stopping we want to restore the best model
         if self.early_stopping and self.early_stopping.save_dir:
             logger.info("Restoring best model so far from {}".format(self.early_stopping.save_dir))
-            self.model = AdaptiveModel.load(self.early_stopping.save_dir, self.device)
+            model_class = type(self.model)
+            self.model = model_class.load(self.early_stopping.save_dir, self.device)
```

Comment:
Instead of these two lines, I suggest to use the following instead:
`self.model = self.model.load(self.early_stopping.save_dir, self.device)`

Although self.model is an instance of BiAdaptiveModel (or AdaptiveModel or ...), we can call `self.model.load` to call the class method. If you want to read a bit about the background, have a look here: https://stackoverflow.com/questions/692040/is-it-bad-form-to-call-a-classmethod-as-a-method-from-an-instance
Another alternative would be to use `self.model.__class__.load` to make it more explicit that we are calling a class method here.